editor/components/post-permalink/editor.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { withDispatch } from '@wordpress/data';
+import { Component, compose } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+import { getEditedPostAttribute } from '../../store/selectors';
+
+class PostPermalinkEditor extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			editedPostName: '',
+		};
+
+		this.onSavePermalink = this.onSavePermalink.bind( this );
+	}
+
+	/**
+	 * Returns the permalink parts to populate the form.
+	 *
+	 * @return {Object} The prefix, suffix, and postName for the form.
+	 */
+	getPermalinkParts() {
+		const [ template, postName ] = this.props.samplePermalinkData;
+		const [ samplePermalinkPrefix, samplePermalinkSuffix ] = template.split( /%(?:postname|pagename)%/ );
+
+		return {
+			samplePermalinkPrefix,
+			samplePermalinkSuffix,
+			postName,
+		};
+	}
+
+	onSavePermalink() {
```

Comment:
Should we `event.preventDefault` anywhere in here to prevent the default behavior of the form? Not sure if this is an issue in any browser, but seems like something where it could be reloading the page.